plugins/com.google.cloud.tools.eclipse.appengine.newproject.test/src/com/google/cloud/tools/eclipse/appengine/newproject/XmlTest.java

```diff
@@ -0,0 +1,53 @@
+package com.google.cloud.tools.eclipse.newproject.whitelist;
+
+import java.io.File;
+import java.io.IOException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.core.runtime.IExtensionPoint;
+import org.eclipse.core.runtime.IExtensionRegistry;
+import org.eclipse.core.runtime.RegistryFactory;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+// todo: extract common code into a utility class
+public class XmlTest {
+    
+  private Document document;
+	
+  @Before
+  public void setUp() throws ParserConfigurationException, IOException, SAXException {
+    DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+    factory.setNamespaceAware(true);
+    DocumentBuilder builder = factory.newDocumentBuilder();
+    document = builder.parse(
```

Comment:
I'm not sure we can assume the test is run from the project's working directory, but it seems to work and we can fix that later as required.

> Tried it but "The import org.osgi cannot be resolved".

Need to `Import-Package: org.osgi.framework`.